src/Indigo/Contracts/HomebaseLite/Impl/Admin.hs

```diff
@@ -15,53 +16,86 @@ import Indigo hiding ((*))
     Takes in input the address of the new `admin` candidate.
-    Note: for security reasons the transfer isn't complete until the new admin candidate calls `accept_admin`.
-    If there is already a candidate, another call to this entrypoint will replace it.
-    If there is already a candidate, calling this with the `admin` address will invalidate the current candidate.
+    Note: for security reasons the transfer isn't complete until the new admin
+    candidate calls `accept_admin`.
+    If there is already a candidate, another call to this entrypoint will
+    replace it.
+    If there is already a candidate, calling this with the `admin` address will
+    invalidate the current candidate.
     |]
-  pass
-
+  checkSenderIsAdmin
+  storage =: storage !! (#sAdminCandidate, newAdmin)
 
-acceptAdmin :: tp :~> () => IndigoEntrypoint tp
+acceptAdmin :: HomebaseLiteEntrypoint ()
 acceptAdmin _ = do
   description [itu|
-    Called by an `admin` candidate (see `set_admin`) to complete the transfer of the role.
-    Takes no input.
+    Called by an `admin` candidate (see `set_admin`) to complete the transfer of
+    the role. Takes no input.
     |]
-  pass
+  candidate <- new$ storage #! #sAdminCandidate
+  when (sender /= candidate) do
+    failCustom_ @() #senderIsNotAdminCandidate
+  storage =: storage !! (#sAdmin, candidate)
 
-[errorDoc| "senderIsNotAdminCandidate" exception "Fail when the sender is not the current admin
-candidate"|]
+[errorDoc| "senderIsNotAdminCandidate" exception "The sender is not the current admin candidate"|]
 
-addMaintainers :: IndigoEntrypoint tp
-addMaintainers _ = do
+addMaintainers :: HomebaseLiteEntrypoint [Address]
+addMaintainers newMaintainers = do
   description [itu|
     Takes as input a list of addresses that will become `maintainer`s.
     All the given addresses receive the `maintainer` role.
     If a given address already has a `maintainer` role or is present in the list
     more than once, this call is a no-op.
     |]
-  pass
+  checkSenderIsAdmin
+  modifyMaintainres (\l r -> l +: (r, ())) newMaintainers
 
-removeMaintainers :: IndigoEntrypoint tp
-removeMaintainers _ = do
+removeMaintainers :: HomebaseLiteEntrypoint [Address]
+removeMaintainers maintainersToRemove = do
   description [itu|
     Takes as input a list of addresses that will lose the `maintainer` role.
     All the given addresses lose the `maintainer` role.
     If a given addresses is not a `maintainer` this call is a no-op.
     |]
-  pass
+  checkSenderIsAdmin
+  modifyMaintainres (-:) maintainersToRemove
 
-configure :: IndigoEntrypoint tp
-configure _ = do
+modifyMaintainres
+  :: (HasStorage Storage, tp :~> [Address])
+  => (Var (BigMap Address ()) -> Var Address -> Expr (BigMap Address ()))
+  -> tp
+  -> IndigoM ()
+modifyMaintainres modf list = do
+  maints <- new$ storage #! #sMaintainers
+  forEach list \maint -> do
+    maints =: maints `modf` maint
+  storage =: storage !! (#sMaintainers, maints)
```

Comment:
similarly, this could use `updateStorageField`:
```suggestion
modifyMaintainres modf list =
  updateStorageField @Storage #sMaintainers $ \maints -> do
    forEach list \maint -> do
      maints =: maints `modf` maint
    return maints
```